WordPress/MediaEditor/MediaEditor.swift

```diff
@@ -1,103 +1,67 @@
-    }
-
-    public var rotateCounterclockwiseButton: UIButton? {
-        return cropViewController.toolbar.rotateCounterclockwiseButton
-    }
-
-    public var rotateCounterclockwiseButtonHidden: Bool {
-        get {
-            return cropViewController.toolbar.rotateCounterclockwiseButtonHidden
-        }
+    private(set) var currentCapability: MediaEditorCapability?
 
-        set {
-            cropViewController.toolbar.rotateCounterclockwiseButtonHidden = newValue
+    public var styles: MediaEditorStyles = [:] {
+        didSet {
+            currentCapability?.apply(styles: styles)
         }
     }
 
-    public var rotateClockwiseButtonHidden: Bool {
-        get {
-            return cropViewController.toolbar.rotateClockwiseButtonHidden
-        }
-
-        set {
-            cropViewController.toolbar.rotateClockwiseButtonHidden = newValue
-        }
+    init(_ image: UIImage) {
+        self.image = image
+        hub = MediaEditorHub()
+        super.init(rootViewController: hub)
+        presentIfSingleImageAndCapability()
+        modalTransitionStyle = .crossDissolve
+        modalPresentationStyle = .fullScreen
+        navigationBar.isHidden = true
     }
 
-    public init(cropViewControllerFactory: @escaping (UIImage) -> TOCropViewController = TOCropViewController.init, image: UIImage) {
-        self.cropViewControllerFactory = cropViewControllerFactory
-        self.image = image
-        super.init()
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
 
     public func edit(from viewController: UIViewController? = nil, onFinishEditing: @escaping (UIImage, [MediaEditorOperation]) -> (), onCancel: (() -> ())? = nil) {
         self.onFinishEditing = onFinishEditing
         self.onCancel = onCancel
-        cropViewController.delegate = self
-        viewController?.present(cropViewController, animated: true)
+        viewController?.present(self, animated: true)
     }
 
-    public func dismiss(animated: Bool, completion: (() -> ())? = nil) {
-        cropViewController.dismiss(animated: animated, completion: completion)
-    }
+    func presentIfSingleImageAndCapability() {
+        guard let _ = image, Self.capabilities.count == 1, let capabilityEntity = Self.capabilities.first else {
+            return
+        }
 
-    private func releaseCallbacks() {
-        onFinishEditing = nil
-        onCancel = nil
+        present(capability: capabilityEntity)
     }
 
-}
-
-// MARK: - TOCropViewControllerDelegate
-
-extension MediaEditor: TOCropViewControllerDelegate {
-
-    public func cropViewController(_ cropViewController: TOCropViewController, didCropTo image: UIImage, with cropRect: CGRect, angle: Int) {
-        let actions = cropViewController.actions
-        onFinishEditing?(image, actions)
-        releaseCallbacks()
-    }
+    private func present(capability capabilityEntity: MediaEditorCapability.Type) {
```

Comment:
Are we ever going to present without an image? Can we pass the image to it?
We are already checking that the image is not nil in the method above (`presentIfSingleImageAndCapability`)

Reply:
Very good feedback, thank you!

Done in ab70c27d0f06592f9f88fff80c932aabf178b33b